axum-core/src/body.rs

```diff
@@ -3,7 +3,10 @@
 use crate::{BoxError, Error};
 use bytes::Bytes;
 use bytes::{Buf, BufMut};
-use http_body::Body;
+use http_body::Body as _;
+
+#[doc(no_inline)]
+pub use hyper::Body;
```

Comment:
Note this makes hyper a public dependency of axum-core. I think thats a little unfortunate but its required to make `impl FromRequest for Request<hyper::Body>` work. I think thats an okay trade-off.

Reply:
It also does mean that any crate that depends on axum-core now also depends on hyper. Thats pretty unfortunate 🤔 But yeah not sure there is a way around it.

Reply:
One solution would be to keep the type param on `FromRequest` and `RequestParts` but that would mean they get three type params in 0.6 😬

Reply:
Perhaps its also worth considering adding our own body type to `axum-core`, to avoid the dependency on hyper. I believe hyper's `Body` will be moved into hyper-utils for 1.0 and we don't wanna have a public dependency on hyper-utils. So we'll probably need our own body type eventually.